source/use-cases/sensitive-data-encryption.txt

```diff
@@ -0,0 +1,312 @@
+=========================
+Sensitive Data Encryption
+=========================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Introduction
+------------
+
+Cras sed rutrum nisi, vitae tristique risus. Aliquam erat volutpat.
+Donec auctor nisl eget ornare accumsan. Pellentesque lectus tortor,
+consectetur molestie pellentesque ac, blandit non libero.
+
+Problem
+~~~~~~~
+
+Pellentesque non mi et est congue mattis pretium sit amet mi. Praesent
+nec tortor nec arcu molestie dapibus. Phasellus at nunc ut quam molestie
+finibus. Suspendisse finibus lectus eget nulla eleifend, vel consequat
+leo rhoncus. Phasellus lacinia tortor id aliquet condimentum. Morbi
+sodales est a leo rhoncus, non pretium neque dignissim.
+
+Solution
+~~~~~~~~
+
+Praesent sit amet elit pretium mi molestie porta a sit amet ipsum. Sed
+sollicitudin purus et mi blandit sodales.
+
+- Deterministic vs Non-Deterministic
+
+Procedure
+---------
+
+Requirements
+~~~~~~~~~~~~
+
+- Point to setup and install directions
+- Nullam imperdiet lorem vitae vulputate lacinia.
+- Donec eget velit tincidunt, gravida diam ac, efficitur lacus.
+
+A. Create a Local Master Key
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Aenean eu consequat lorem. Ut posuere est sed sodales pharetra. Cras
+volutpat, massa laoreet varius dictum, leo odio porttitor ante, nec
+auctor tortor orci et mi. Maecenas tempor, lacus vehicula molestie
+pulvinar, ante eros faucibus odio, sed consequat quam tellus vel arcu.
+Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere
+cubilia Curae; Nunc non interdum purus, ultricies laoreet tortor.
+
+B. Define a JSON Schema
+~~~~~~~~~~~~~~~~~~~~~~~
+
+MongoDB uses JSON Schema to to validate data which is written to a collection.
```

Comment:
I wonder if it would be helpful to link to the MongoDB JSON Schema Validation page (https://docs.mongodb.com/manual/core/schema-validation/) since the reader might not be familiar with what it is.

Reply:
Nitpick: it is not completely clear whether "written to a collection" refers to data that has already been written (which JSON Schema Validation does not automatically do) or that it happens when data is written. More specificity might be useful, e.g.
```
In MongoDB, we can create and specify a JSON Schema on a collection to validate insert and update operations.
```

Reply:
I added a reference to the schema validation link in that sentence, and I completely agree with the second comment as well. Both suggestions have been implemented and pushed in the most recent commit.